include/deal.II/cgal/polygon_2.h

```diff
@@ -0,0 +1,139 @@
+#ifndef dealii_cgal_polygon_2_h
+#define dealii_cgal_polygon_2_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/fe/mapping.h>
+
+#include <deal.II/grid/tria.h>
+
+#ifdef DEAL_II_WITH_CGAL
+#  include <deal.II/cgal/point_conversion.h>
+#  include <CGAL/Polygon_2.h>
+
+
+//only for compute boolean operation
+#   include <CGAL/Boolean_set_operations_2.h>
+#   include <CGAL/Polygon_with_holes_2.h>
+#   include <deal.II/cgal/utilities.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace CGALWrappers
+{
+  using K         = CGAL::Exact_predicates_exact_constructions_kernel;
+
+  using CGALPoint2 = CGAL::Point_2<K>;
+  using CGALPolygon = CGAL::Polygon_2<K>;
+
+  void dealii_cell_to_cgal_polygon(
+    const typename Triangulation<2, 2>::cell_iterator &cell,
+    const Mapping<2, 2>                               &mapping,
+    CGALPolygon                                       &polygon)
+  {
+    //Note that this is accounting for curved boundary
+    // not sure if needed but fitted geometry might be curved!
+    // if not needed mapping object is obsolete then use cell->vertex()
+    const auto &vertices = mapping.get_vertices(cell);
+    polygon.clear();
+    if(cell->reference_cell() == ReferenceCells::Triangle)
+    {
+      polygon.push_back(CGALWrappers::dealii_point_to_cgal_point<CGALPoint2,2>(vertices[0]));
```

Comment:
```suggestion
      polygon.push_back(dealii_point_to_cgal_point<CGALPoint2,2>(vertices[0]));
```
We are inside the `CGALWrappers` namespace.